Samples/HelloWorld/OrleansClient/Program.cs

```diff
@@ -16,8 +16,27 @@ public class Program
         static int Main(string[] args)
         {
             var config = ClientConfiguration.LocalhostSilo();
-            const int initializeAttemptsBeforeFailing = 5;
-            int attemp = 0;
+            try
+            {
+                InitializeWithRetries(config, initializeAttemptsBeforeFailing: 5);
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"Orleans client initialization failed failed due to {ex}");
+
+                Console.ReadLine();
+                return 1;
+            }
+
+            DoClientWork().Wait();
+            Console.WriteLine("Press Enter to terminate...");
+            Console.ReadLine();
+            return 0;
+        }
+
+        private static void InitializeWithRetries(ClientConfiguration config, int initializeAttemptsBeforeFailing)
+        {
+            int attempt = 0;
             while (true)
             {
```

Comment:
I did not change how it worked. Nevertheless, if the exception is of a different type, then it will not be caught and would bubble up, breaking out of the lool, so there is no chance for an infinite loop.
I would not like to catch the general `Exception` type.